pyatlan/cache/atlan_tag_cache.py

```diff
@@ -64,8 +65,10 @@ def _refresh_cache(self) -> None:
         Refreshes the cache of Atlan tags by requesting the full set of Atlan tags from Atlan.
         """
         response = self.provider.get_typedefs(
-            type_category=AtlanTypeCategory.CLASSIFICATION
+            type_category=[AtlanTypeCategory.CLASSIFICATION, AtlanTypeCategory.STRUCT]
```

Comment:
Why not raise the exception in the AtlanClient if all the lists are empty? That would remove some duplication of code.

Reply:
Each cache is only retrieving the minimal info it requires — so if some code _only_ needs to deal with tags, there's no need to retrieve all the custom metadata definitions (and vice versa). If we moved this up to the client level, then we'd need to retrieve the entire set of all type definitions, irrespective of whether someone is actually going to make use of them in their code or not (can add significant, unnecessary delay).

Reply:
I'm not suggesting retrieving all the metadata but if all the lists are empty, which I believe is what we generally see, that's a problem.